src/main/java/cn/nukkit/block/BlockDoor.java

```diff
@@ -33,26 +35,29 @@
  */
 public abstract class BlockDoor extends BlockTransparentMeta implements Faceable {
     private static final double THICKNESS = 3.0 /16;
-    
+
+    // no idea how to make this better :d
+    private static final List<Location> manualOverrides = new ArrayList<>();
```

Comment:
After reading the whole diff for `BlockDoor`, I'm still lost on what this `manualOverrides` field do. Can you clarify it for me?

Reply:
It contains a list of positions of Doors, which have been opened by hand (by a player). It is used to detect on redstone update, if the door should be closed if redstone is off on the update, previously the door always closed, when placing an unpowered redstone at the door, this fixes it and gives the vanilla behavior.

Reply:
I added a comment describing it.